docs_src/optimizations.md

```diff
@@ -324,6 +324,23 @@ additional selector which is the logical NOR of all of the original `Select`
 selector bits is appended to the `OneHotSelect` selector and the respective case
 is the value selected when all selector bits are zero (`case_0` in the diagram).
 
+### Simplifying select arms
+
+Within any given arm of a `Select` multiplexer, it is safe to assume that the
+selector has the specific value required to select that arm.
```

Comment:
Maybe add something like: this assumption is safe because in the event that the selector has a different value the arm is dead.